zyngui/zynthian_gui_engine.py

```diff
@@ -91,38 +100,61 @@ def __init__(self):
 	def set_engine_type(self, etype):
 		self.engine_type = etype
 		self.midi_chan = None
-
+		self.init_engine_info()
 
 	def set_fxchain_mode(self, midi_chan):
 		self.engine_type = "Audio Effect"
 		self.midi_chan = midi_chan
-
+		self.init_engine_info()
 
 	def set_midichain_mode(self, midi_chan):
 		self.engine_type = "MIDI Tool"
 		self.midi_chan = midi_chan
+		self.init_engine_info()
 
+	def fill_list_item(self, en, info):
+		# For some engines, check if needed channels are free ...
+		if (en in self.check_channels_engines and
+			not all(chan in self.zyngui.screens['layer'].get_free_midi_chans() for chan in info[3].get_needed_channels())):
+			return
+
+		ei=self.engine_info[en]
+		self.list_data.append((en,len(self.list_data),ei[1],ei[0]))
 
 	def fill_list(self):
 		self.init_engine_info()
-		self.index=0
 		self.list_data=[]
-		i=0
-		for en, info in self.engine_info.items():
-			if (info[4] and (info[2]==self.engine_type or self.engine_type is None) and
-				(en not in self.single_layer_engines or en not in self.zyngines)):
 
-				# For some engines, check if needed channels are free ...
-				if (en in self.check_channels_engines and
-					not all(chan in self.zyngui.screens['layer'].get_free_midi_chans() for chan in info[3].get_needed_channels())):
-						continue
+		# Sort category headings, but headings starting with "Zynthian" are shown first
+		for cat, infos in sorted(self.engine_info_categorized.items(), key = lambda kv:
+			                 "! {}".format(kv[0]) if kv[0].startswith("Zynthian") else kv[0]):
+			self.list_data.append((None,len(self.list_data),"  {}".format(cat)))
+			# Sort the list of plugins, unless it's the built-in list (which has a fixed order)
+			if cat.startswith("Zynthian"):
+				items = infos.items()
+			else:
+				items = sorted(infos.items(), key = lambda kv: kv[0])
+			for en, info in items:
+				self.fill_list_item(en, info)
 
-				ei=self.engine_info[en]
-				self.list_data.append((en,i,ei[1],ei[0]))
-				i=i+1
+		if (len(self.list_data) == 0):
+			self.list_data.append((None,len(self.list_data),"Please use webconf to enable plugins."))
```

Comment:
Actually, I'll see if I can just show an actual URL here; early users may not recognize the term "Webconf" either.

Reply:
New users shouldn't see this message, as pre-built images should have a pre-configured selection of engines for every type (i think MIDI tool is currently empty, although this will change very soon).